test/github-issues/9944/issue-9944.ts

```diff
@@ -0,0 +1,42 @@
+import "reflect-metadata"
+import {
+    createTestingConnections,
+    closeTestingConnections,
+    reloadTestingDatabases,
+} from "../../utils/test-utils"
+import { DataSource } from "../../../src/data-source/DataSource"
+import { expect } from "chai"
+import { A } from "./entity/A"
+import { B } from "./entity/B"
+import { C } from "./entity/C"
+
+describe("github issues > #9944 Alias Issue With Nested Entity And Relations", () => {
+    let dataSources: DataSource[]
+    before(
+        async () =>
+            (dataSources = await createTestingConnections({
+                entities: [__dirname + "/entity/*{.js,.ts}"],
+                schemaCreate: true,
+                dropSchema: true,
+                relationLoadStrategy: "query",
+                enabledDrivers: ["mysql"],
+            })),
+    )
+    beforeEach(() => reloadTestingDatabases(dataSources))
+    after(() => closeTestingConnections(dataSources))
+
+    it("Validate correct loading of eager, nested relations", () =>
+        Promise.all(
+            dataSources.map(async (dataSource) => {
+                const aEntity = new A()
+                aEntity.b = new B()
+                aEntity.b.cs = [new C()]
+                await dataSource.manager.save(aEntity)
+
+                const [fetchedA] = await dataSource.manager.find(A)
+                expect(fetchedA.b).exist // Validates correct relation A.b is loaded
+                expect(fetchedA.b.cs).exist // Validates correct relation A.b.cs is loaded
+                expect(fetchedA.b.cs[0]).exist
```

Comment:
Before the fix, running this test resulted in the `TypeORMError: "A__b" alias was not found. Maybe you forgot to join it?`

With the fix it runs perfectly. I jut added these three assertions to make sure that data from nested relations was indeed loaded correctly